.github/workflows/test.yaml

```diff
@@ -21,14 +21,14 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v4
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```

Comment:
why do we need to pin commits like this?

Reply:
There is a [nice article](https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash) about this. Here are the three main reasons they give

>  They guarantee that checking and deployment are performed with the same software, minimizing deployment risks, simplifying debugging, and enabling reproducibility.

> They can help mitigate compromised dependencies from compromising the project’s security. By evaluating the pinned dependency and being confident that it is not compromised, you can prevent the use of a later version that may be compromised.

> Pinned dependencies are a way to counter dependency confusion (or substitution) attacks. In these attacks, an application uses multiple feeds to acquire software packages (a “hybrid configuration”), and attackers trick the user into using a malicious package from an unexpected feed.

Reply:
I am not referring to the dependencies in the build, but to the pinning of github action versions. We don't want to maintain the pinning of such gory details.  Why is that required?

Reply:
I'm referring to GitHub actions. You won't maintain them. Dependabot can

Reply:
I've read the article. I understand better now. My concern is that we have  13+ repos to maintain.

Reply:
I guess you're worried about the number of PR you'll have to validate?

You can change the frequency and also group the dependencies to reduce this number.

Reply:
Can you imagine give this a try for a certain time. Identify and fix the irritant if there are and roll it back if it's a fail

Reply:
Yes that could be possible. What is your own feedback about using this technique in other projects?

Reply:
It quite practical and flexible enough to fit with a variety of practice.
Some teams prefer daily checking some other monthly depending on the frequency they publish release or follow another logic. 

Depending on the amount of dependencies you've got in the project it is some time preferable to use renovate which is way more configurable than dependabot.

But for most of the use case dependabot is good enough.

As it creates a PR, it executes your pipeline with all your tests. Most of the time you just click on merge because there is nothing to fix.

You can can even have a workflow that automatically validate those merge requests if it passes the pipeline.

Reply:
ok let's give at try. some other contributors have pushed forward to use dependabot in our other repos.
I am fine with that but I think I'll reduce the frequency or configure some kind . auto merge